src/System.IO.Pipelines/System/Buffers/MemoryPool.cs

```diff
@@ -78,6 +78,12 @@ public override OwnedMemory<byte> Rent(int size)
             return block;
         }
 
+        public override void Return(OwnedMemory<byte> buffer)
+        {
+            var block = buffer as MemoryPoolBlock;
+            if (block != null) Return(block);
```

Comment:
Throw?

Reply:
I don't think we need this public, only thing that returns is `OnZeroReferences` of `MemoryPoolBlock`

Reply:
Yeah, I was thinking about it. I will change to throw.